src/catalog/src/tables.rs

```diff
@@ -266,10 +276,9 @@ impl SystemCatalog {
         &self,
         catalog: String,
         schema: String,
-        table_name: String,
         table_id: TableId,
     ) -> crate::error::Result<usize> {
-        let full_table_name = format_full_table_name(&catalog, &schema, &table_name);
+        let full_table_name = format_full_table_name_by_id(&catalog, &schema, &table_id);
```

Comment:
Yes, the system catalog must be updated when a table is renamed. In the system catalog schema, key_column_len is 3, while other columns are not allowed to be null. It will cause panic when creating the RecordBatch. To delete a row from the system catalog table, I modified the constraints of the non-key columns in the system catalog schema to be nullable.

Reply:
What about storing the table name in the JSON body? We could still use `<catalog_name>.<schema_name>.<table_id>` as key so renaming is easy: just re-insert the entry with the new table name.

> In the system catalog schema, key_column_len is 3, while other columns are not allowed to be null. It will cause panic when creating the RecordBatch

Oops, this should be a bug, I would fix it. I have created issue #848  for it

Reply:
> What about storing the table name in the JSON body? We could still use `<catalog_name>.<schema_name>.<table_id>` as key so renaming is easy: just re-insert the entry with the new table name.

I tested it, and it still shows the original table name after restarting the instance. It seems that the table entry in the system catalog table has not been overwritten. Is there something I missed?

Reply:
> What about storing the table name in the JSON body? We could still use `<catalog_name>.<schema_name>.<table_id>` as key so renaming is easy: just re-insert the entry with the new table name.

fixed. PTAL